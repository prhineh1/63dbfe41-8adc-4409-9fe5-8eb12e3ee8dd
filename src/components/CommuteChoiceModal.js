import React from 'react';
import Modal from 'react-modal';
import ReactAnimatedWeather from 'react-animated-weather';

const CommuteChoiceModal = (props) => {
  const modeDeclaration = (
    props.weather.mode === 'metro' ? 'You should take the metro.' :
    "You're all clear; bike it."
  );
  return (
    <Modal
      isOpen={props.isOpen} // propTypes.bool.isRequired
      onRequestClose={props.toggleModal} //propTypes.func
      contentLabel='The Weather and Your Commute' // adds aria-label to the modal
      closeTimeoutMS={200}
    >
      <h4>{modeDeclaration}</h4>
      <div>
        <ReactAnimatedWeather
          icon={props.weather.icon}
        />
        <span>
          {props.weather.summary} at  
          {props.weather.temperature} with a 
          {props.weather.precipProbability * 100}% chance of precipitation
        </span>
      </div>
      <button onClick={props.toggleModal}>Ok</button>
    </Modal>
  )
}

export default CommuteChoiceModal;
