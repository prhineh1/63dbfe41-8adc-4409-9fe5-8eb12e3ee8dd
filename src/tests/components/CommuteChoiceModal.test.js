import React from 'react';
import { shallow, setProps } from 'enzyme';
import CommuteChoiceModal from '../../components/CommuteChoiceModal';
import data from '../fixtures/weatherData';

let wrapper, toggleModal;

beforeEach(() => {
  toggleModal = jest.fn();
  wrapper = shallow(<CommuteChoiceModal isOpen toggleModal={toggleModal} weather={data[1]} />);
});

test('should render modal when with bike commute', () => {
  const bike = wrapper.setProps({ commute: false });
  expect(bike).toMatchSnapshot();
});

test('should render modal when with metro commute', () => {
  const bike = wrapper.setProps({ commute: true });
  expect(bike).toMatchSnapshot();
});

test('should handle toggleModal', () => {
  wrapper.find('button').simulate('click');
  expect(toggleModal).toHaveBeenCalled();
});