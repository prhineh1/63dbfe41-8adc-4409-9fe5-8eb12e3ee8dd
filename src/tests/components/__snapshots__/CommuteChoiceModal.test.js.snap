// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render modal when with bike commute 1`] = `
<Modal
  ariaHideApp={true}
  bodyOpenClassName="ReactModal__Body--open"
  closeTimeoutMS={200}
  contentLabel="The Weather and Your Commute"
  isOpen={true}
  onRequestClose={[Function]}
  parentSelector={[Function]}
  portalClassName="ReactModalPortal"
  shouldCloseOnEsc={true}
  shouldCloseOnOverlayClick={true}
  shouldFocusAfterRender={true}
  shouldReturnFocusAfterClose={true}
>
  <h4>
    You're all clear; bike it.
  </h4>
  <div>
    <ReactAnimatedWeather
      animate={true}
      color="black"
      icon="CLEAR_DAY"
      size={64}
    />
    <span>
      clear skies
       at
      45
       with a
      0
      % chance of precipitation
    </span>
  </div>
  <button
    onClick={[Function]}
  >
    Ok
  </button>
</Modal>
`;

exports[`should render modal when with metro commute 1`] = `
<Modal
  ariaHideApp={true}
  bodyOpenClassName="ReactModal__Body--open"
  closeTimeoutMS={200}
  contentLabel="The Weather and Your Commute"
  isOpen={true}
  onRequestClose={[Function]}
  parentSelector={[Function]}
  portalClassName="ReactModalPortal"
  shouldCloseOnEsc={true}
  shouldCloseOnOverlayClick={true}
  shouldFocusAfterRender={true}
  shouldReturnFocusAfterClose={true}
>
  <h4>
    You should take the metro.
  </h4>
  <div>
    <ReactAnimatedWeather
      animate={true}
      color="black"
      icon="CLEAR_DAY"
      size={64}
    />
    <span>
      clear skies
       at
      45
       with a
      0
      % chance of precipitation
    </span>
  </div>
  <button
    onClick={[Function]}
  >
    Ok
  </button>
</Modal>
`;
